### DISCUSSION TITLE
Standalone usage/Binary distribution

### DISCUSSION THREAD

**Discussion by AngelMunoz** [swc-project/swc#2290] [Q&A] [answered]
Hello everyone Thank you for this amazing project

Is there a way to use swc without node? 

I currently use esbuild binaries to do some js processing in a dev tool I'm working with, but I'd like to support swc in the future, However it looks like the CLI is meant with node in mind. 

If anyone can point me to the correct resources to investigate this further it would be great! thanks in advance

**Comment by kdy1** [comment]
Are you going to use swc from rust?
You can just depend on it then.

There's https://github.com/swc-project/swc/issues/1589, but I don't have enough time to do it.

**Reply by AngelMunoz** [reply]
I'm trying to use it from .NET, what I currently do with esbuild is that my tool downloads it into a known location in disk then just execute the process myself, I was hoping doing the same with swc

**Reply by kdy1** [reply]
You can create bindings, but I don't think it will worth the effort.

**Comment by chrishecker** [comment]
Bumping this.  I was kind of shocked to find literally all the typescript compilers require node.  As somebody who just wanted to play around with typescript for just normal js on web pages, I don't need node, and it's weird there isn't just a single simple executable to do the transpiling.  Seems like swc is the closest to getting there since it's in rust...I might take a look at compiling it standalone.

**Reply by kwonoj** [reply]
We have https://github.com/swc-project/swc/discussions/3859 .

**Reply by chrishecker** [reply]
Ah, I was wondering why I was seeing references to swc.exe, I had just started building (and found the released source uses rust nightly features?!)   `error[E0554]: #![feature] may not be used on the stable release channel`

Anyway, I'll take a look, thanks!

**Reply by chrishecker** [reply]
Okay, I got this working and it's exactly what I was looking for in terms of standalone-ness (there is some weird bug with running in an emacs shell on windows, it thinks it's getting files from stdin and stdout even with command line parms, weird...but it works fine from the command line), but I think I must have misunderstood what swc does and doesn't do...it doesn't seem to want to typecheck a .ts file, which is what I was looking for.  Am I being dumb and missing something?  Ah, the all important phrase on the typescript page that I skimmed over: 
> Each of these projects handle the file conversion, but do not handle the type-checking aspects of the TypeScript compiler. 

Oops.  Sigh.  Oh well, I will go back to being shocked.

**Comment by kwonoj** [comment, marked as answer]
Please refer https://github.com/swc-project/swc/discussions/3859
